src/Microsoft.SqlTools.ServiceLayer/Management/Common/DataContainer.cs

```diff
@@ -1081,6 +1081,12 @@ private void Dispose(bool disposing)
 
                     if (disposing)
                     {
+                        // Disconnect our own connection, since disposing sqlCiWithConnection may not catch it
+                        if (ownConnection && this.serverConnection != null)
+                        {
+                            this.serverConnection.Disconnect();
```

Comment:
Any reason not to use dispose instead? And should probably null this out afterwards like the others do.